src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/LLVMObjectWriter.cs

```diff
@@ -39,18 +41,15 @@ internal sealed unsafe class LLVMObjectWriter
         // Path to the bitcode file we're emitting.
         private readonly string _objectFilePath;
 
-        // Raw data emitted for the current object node.
-        private ArrayBuilder<byte> _currentObjectData;
-
-        // References (pointers) to symbols the current object node contains (and thus depends on).
-        private Dictionary<int, SymbolRefData> _currentObjectSymbolRefs = new();
-
-        // Data to be emitted as LLVM bitcode after all of the object nodes have been processed.
-        private readonly List<ObjectNodeDataEmission> _dataToFill = new();
+        // Used for writing mangled names together with Utf8Name.
+        private readonly Utf8StringBuilder _utf8StringBuilder = new Utf8StringBuilder();
 
         // List of global values to be kept alive via @llvm.used.
         private readonly List<LLVMValueRef> _keepAliveList = new();
 
+        // Data emitted for the current object node. Initial capacity chosen to be the size of the largest node in a small program.
+        private LLVMValueRef[] _currentObjectData = new LLVMValueRef[100_000];
```

Comment:
This largest object node was `__embedded_metadata` aka `MetadataNode` aka reflection info.